protocols/luci-proto-yggdrasil/root/usr/libexec/rpcd/luci.yggdrasil

```diff
@@ -16,12 +16,12 @@ case "$1" in
 		case "$2" in
 			generateKeyPair)
 				json_load "$(yggdrasil -genconf -json)"
-				json_get_vars PublicKey PrivateKey
+				json_get_vars PrivateKey
 				json_cleanup
 				json_init
 				json_add_object "keys"
 				json_add_string "priv" "$PrivateKey"
-				json_add_string "pub" "$PublicKey"
+				json_add_string "pub" "${PrivateKey:64}"
```

Comment:
So my eyes are naïve here but the public half is the private key truncated at 64 bytes? If that's the way things are, that's fine. 

But this isn't the more cpu intensive genkeys variation is it? https://yggdrasil-network.github.io/configuration.html

My concern here is whether the GUI would time out if the operation takes a long time.

Reply:
this is whats going on.
we would need to build and package genkeys. i would like to see a 'generate keys' button with possibly some input (a checkbox or some sort of input/dropdown) to try and find a key within say a 1 to 3 (or whatever) timeout duration.

the current -genkeys is (i think) first find first use.

we would need to make some sort of non-blocking operation to poll for the existence of /tmp/genkeys.txt (or something).. when the end of the duration is hit. i last checked and we do not have a `timeout` with busybox.. so... genkeys will max out all logical cpu cores when executed. i think we could patch the genkeys.go file with some router-friendly parameters eg; -timeout and -save-last-key-to-file

fair to say this is out of scope of LuCI for now.. i'll see if i cant patch/upstream something worth using.

if we get this going, how do we do it in LuCI? is there a non-blocking way to go about this?

as a user, i wouldnt care if i did 'generate key' and 'key generation timeout=10min' ... if i could set it and come back in 10 min to see available keys (or last key it found) .. sure i'll take that.

Reply:
> this is whats going on. we would need to build and package genkeys. i would like to see a 'generate keys' button with possibly some input (a checkbox or some sort of input/dropdown) to try and find a key within say a 1 to 3 (or whatever) timeout duration.
Most router CPUs would probably not be good at iterating through many keypairs. If you were to mine it in Javascript, though...

Reply:
In terms of protocol design in yggdrasil, it seems... terrible. I'm just making a drive-by comment here, rather than attempting to appreciate the reasoning behind this choice. 

On the surface, this seems like a valid suggestion, but... Does there exist an Ed25519 JS implementation to even go this route? Waiting for 10 minutes until a valid configuration can be built does not seem very user-friendly. Websites are dead in the water if they don't respond within 7 seconds.

Reply:
key generation is performed only once when a configuration is generated.

Reply:
> the current yggdrasil -genconf is (i think) first find and first use.

my 2 cents as an observer (as well as longtime yggdrasil user and occasional openwrt contributor) is that it would be beneficial to get this PR merged with just the "first find and first use" key generation, as that is what most people should/would need.